syntax = "proto3";
package CWAConfig;

// taken from:
// https://github.com/corona-warn-app/cwa-app-android/tree/main/Server-Protocol-Buffer/src/main/proto/internal/stats

message CardHeader {
  
  // ID of the card; must be unique across all types of cards
  // fed into cardIdSequence
  int32 cardId = 1;

  // UNIX timestamp
  int64 updatedAt = 2;
}

message KeyFigureCard {

  CardHeader header = 1;

  repeated KeyFigure keyFigures = 2;
}

message KeyFigure {

  enum Rank {
    UNSPECIFIED_RANK = 0;
    PRIMARY = 1;
    SECONDARY = 2;
    TERTIARY = 3;
  }

  enum Trend {
    UNSPECIFIED_TREND = 0;
    STABLE = 1;
    INCREASING = 2;
    DECREASING = 3;
  }

  enum TrendSemantic {
    UNSPECIFIED_TREND_SEMANTIC = 0;
    NEUTRAL = 1;
    POSITIVE = 2;
    NEGATIVE = 3;
  }

  Rank rank = 1;
  double value = 2;
  int32 decimals = 3; // zero by default; formatted as integer then
  Trend trend = 4;
  TrendSemantic trendSemantic = 5;
}

message Statistics {

  // The sequence in which cards are displayed
  // remove a cardId here to hide a card from the app
  repeated int32 cardIdSequence = 1;

  repeated KeyFigureCard keyFigureCards = 2;
}
